client/device/cloudftd/cloudftdonboarding/create_test.go

```diff
@@ -0,0 +1,425 @@
+			},
+		},
+		{
+			testName: "error when read fmc domain info failed",
+			input:    cloudftdonboarding.NewCreateInput(ftdUid),
+			setupFunc: func() {
+				ReadFmcIsSuccessful(true)
+				ReadFmcDomainInfoIsSuccessful(false)
+				ReadApiTokenInfoIsSuccessful(true)
+				CreateSystemApiTokenIsSuccessful(true)
+				ReadFtdMetadataIsSuccessful(true)
+				CreateFmcDeviceRecordIsSuccessful(true)
+				ReadTaskStatusIsSuccessful(true)
+				ReadFtdSpecificDeviceIsSuccessful(true)
+				TriggerRegisterFmcStateMachineSuccess(true)
+				TriggerRegisterFmcStateMachineEndsInDone(true)
+			},
+			assertFunc: func(output *cloudftdonboarding.CreateOutput, err error, t *testing.T) {
+				assert.NotNil(t, err)
+				assert.Nil(t, output)
+			},
+		},
+		{
+			testName: "error when read api token info failed",
+			input:    cloudftdonboarding.NewCreateInput(ftdUid),
+			setupFunc: func() {
+				ReadFmcIsSuccessful(true)
+				ReadFmcDomainInfoIsSuccessful(true)
+				ReadApiTokenInfoIsSuccessful(false)
+				CreateSystemApiTokenIsSuccessful(true)
+				ReadFtdMetadataIsSuccessful(true)
+				CreateFmcDeviceRecordIsSuccessful(true)
+				ReadTaskStatusIsSuccessful(true)
+				ReadFtdSpecificDeviceIsSuccessful(true)
+				TriggerRegisterFmcStateMachineSuccess(true)
+				TriggerRegisterFmcStateMachineEndsInDone(true)
+			},
+			assertFunc: func(output *cloudftdonboarding.CreateOutput, err error, t *testing.T) {
+				assert.NotNil(t, err)
+				assert.Nil(t, output)
+			},
+		},
+		{
+			testName: "error when create system token failed",
+			input:    cloudftdonboarding.NewCreateInput(ftdUid),
+			setupFunc: func() {
+				ReadFmcIsSuccessful(true)
+				ReadFmcDomainInfoIsSuccessful(true)
+				ReadApiTokenInfoIsSuccessful(true)
+				CreateSystemApiTokenIsSuccessful(false)
+				ReadFtdMetadataIsSuccessful(true)
+				CreateFmcDeviceRecordIsSuccessful(true)
+				ReadTaskStatusIsSuccessful(true)
+				ReadFtdSpecificDeviceIsSuccessful(true)
+				TriggerRegisterFmcStateMachineSuccess(true)
+				TriggerRegisterFmcStateMachineEndsInDone(true)
+			},
+			assertFunc: func(output *cloudftdonboarding.CreateOutput, err error, t *testing.T) {
+				assert.NotNil(t, err)
+				assert.Nil(t, output)
+			},
+		},
+		{
+			testName: "error when read FTD metadata failed",
+			input:    cloudftdonboarding.NewCreateInput(ftdUid),
+			setupFunc: func() {
+				ReadFmcIsSuccessful(true)
+				ReadFmcDomainInfoIsSuccessful(true)
+				ReadApiTokenInfoIsSuccessful(true)
+				CreateSystemApiTokenIsSuccessful(true)
+				ReadFtdMetadataIsSuccessful(false)
+				CreateFmcDeviceRecordIsSuccessful(true)
+				ReadTaskStatusIsSuccessful(true)
+				ReadFtdSpecificDeviceIsSuccessful(true)
+				TriggerRegisterFmcStateMachineSuccess(true)
+				TriggerRegisterFmcStateMachineEndsInDone(true)
+			},
+			assertFunc: func(output *cloudftdonboarding.CreateOutput, err error, t *testing.T) {
+				assert.NotNil(t, err)
+				assert.Nil(t, output)
+			},
+		},
+
+		// t.Skip("requires override inner retry config support")
```

Comment:
Please do not commit commented out code.

Reply:
sorry, fixed